romancal/pipeline/exposure_pipeline.py

```diff
@@ -71,66 +75,100 @@ def process(self, input):
             input_filename = None
 
         # open the input file
-        input = rdd.open(input)
-
-        log.debug("Exposure Processing a WFI exposure")
-
-        self.dq_init.suffix = "dq_init"
-        result = self.dq_init(input)
-        if input_filename:
-            result.meta.filename = input_filename
-        result = self.saturation(result)
-
-        # Test for fully saturated data
-        if is_fully_saturated(result):
-            log.info("All pixels are saturated. Returning a zeroed-out image.")
-
-            # Return zeroed-out image file (stopping pipeline)
-            return self.create_fully_saturated_zeroed_image(result)
-
-        result = self.linearity(result)
-        result = self.dark_current(result)
-        result = self.jump(result)
-        result = self.rampfit(result)
-
-        # Test for fully saturated data
-        if "groupdq" in result.keys():
+        file_type = filetype.check(input)
+        asn = None
+        if file_type == "asdf":
+            try:
+                input = rdm.open(input)
+            except TypeError:
+                log.debug("Error opening file:")
+        if file_type == "asn":
+            try:
+                asn = LoadAsLevel2Asn.load(input, basename=self.output_file)
+            except AssociationNotValidError:
+                log.debug("Error opening file:")
```

Comment:
Similarly here - if an association is invalid shouldn't the code exit?

Reply:
Is there a way to make sure it's a Level 2 asn, to prevent running the code on L2 products? Even if this is desirable I'm not asking to be done in this PR.

Reply:
I left all that out. I'd like to address that once rcal-620 and 621 are in. I think that would be a better place to enforce validity.